### PR TITLE
[Feature] Sync for manifests and helm chart

### LINKED ISSUE #446: [Feature] need sync for manifests and helm chart
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/kuberay/issues) and found no similar feature requirement.

### Description

In https://github.com/ray-project/kuberay/blob/76f96bf5a05651611f9a8734c34c2a7f23f54b54/ray-operator/Makefile#L49-L50
we may find there is only crds synced and if there is modification in RBAC, we may forget to sync and may cause some RBAC failure, for example add new rayjobs resource verbs and we forget to update the chart manually.

### Use case

we may use `make helm` to sync all the roles as well as crds.

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by kevin85421**
## Why are these changes needed?

The CRD inconsistency between `ray-operator/config/crd/bases/` and `helm-chart/kuberay-operator/crds/` is possible to cause bugs like #538. Hence, we need to check the consistency in GitHub Actions.

## Related issue number
Closes #446 
#538 

## Checks

#565 is the control group.

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: ray-operator/DEVELOPMENT.md [hunk 1]
```diff
@@ -76,4 +76,26 @@ IMG=kuberay/operator:nightly make deploy
 We have [chart lint tests](https://github.com/ray-project/kuberay/blob/master/.github/workflows/helm-lint.yaml) with Helm v3.4.1 and Helm v3.9.4 on GitHub Actions. We also provide a script to execute the lint tests on your laptop. If you cannot reproduce the errors on GitHub Actions, the possible reason is the different version of Helm. Issue [#537](https://github.com/ray-project/kuberay/issues/537) is an example that some errors only happen in old helm versions.
 
 * Step1: Install `ct` (chart-testing) and related dependencies. See https://github.com/helm/chart-testing for more details.
-* Step2: `./helm-chart/script/chart-test.sh`
\ No newline at end of file
+* Step2: `./helm-chart/script/chart-test.sh`
+
+### Consistency check
+
+We have several [consistency checks](https://github.com/ray-project/kuberay/blob/master/.github/workflows/consistency-check.yaml) on GitHub Actions. There are several files need synchronizations.
```
Comment (DmitriGekhtman) [inline ray-operator/DEVELOPMENT.md:83]:
```suggestion
We have several [consistency checks](https://github.com/ray-project/kuberay/blob/master/.github/workflows/consistency-check.yaml) on GitHub Actions. There are several files which need synchronization.
```

File: .github/workflows/consistency-check.yaml [hunk 2]
```diff
@@ -50,3 +50,29 @@ jobs:
...
+        uses: tj-actions/verify-changed-files@v11.1
+        with:
+          files: |
+            ./helm-chart/kuberay-operator/crds/*.yaml
+      - name: Check changed files
+        if: steps.verify-changed-files.outputs.files_changed == 'true'
+        uses: actions/github-script@v3
+        with:
+          script: |
+            core.setFailed('Please run \"make sync\" to synchronize CRD!')
```
Comment (DmitriGekhtman) [inline .github/workflows/consistency-check.yaml:78]:
```suggestion
            core.setFailed('Please run \"make sync\" to synchronize CRDs!')
```

**Review by DmitriGekhtman — APPROVED** [review]
looks great to me (just a couple of grammar nits),
any thoughts @Jeffwan @scarlet25151?

**Review by scarlet25151 — APPROVED** [review]
LGTM.

File: ray-operator/DEVELOPMENT.md (hunk 1 above)
Comment (kevin85421) [reply ray-operator/DEVELOPMENT.md]: Fixed.

File: .github/workflows/consistency-check.yaml (hunk 2 above)
Comment (kevin85421) [reply .github/workflows/consistency-check.yaml]: Fixed.

**Comment by DmitriGekhtman** [comment]
Nice! Do we still need a PR with some logic to keep RBAC in sync?

**Comment by kevin85421** [comment]
> Nice! Do we still need a PR with some logic to keep RBAC in sync?

Oh, I think we need RBAC synchronization. I will open an issue.

**Pull request merged**